File: app/src/main/kotlin/com/weatherflow/app/data/remote/api/AirQualityApi.kt
package com.weatherflow.app.data.remote.api

import com.weatherflow.app.data.remote.dto.AirQualityDto
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import io.ktor.client.request.parameter

class AirQualityApi(private val client: HttpClient) {

    suspend fun fetchAirQuality(lat: Double, lon: Double): AirQualityDto {
        return client.get("https://air-quality-api.open-meteo.com/v1/air-quality") {
            parameter("latitude", lat)
            parameter("longitude", lon)
            parameter("current", "european_aqi,us_aqi,pm2_5,pm10,ozone")
        }.body()
    }
}

File: weather-flow/src/api/airQuality.api.ts
interface AirQualityResponse {
  current: {
    european_aqi: number | null
    us_aqi: number | null
    pm2_5: number | null
    pm10: number | null
    ozone: number | null
  }
}

export async function fetchAirQuality(lat: number, lon: number): Promise<AirQualityResponse> {
  const params = new URLSearchParams({
    latitude: lat.toString(),
    longitude: lon.toString(),
    current: 'european_aqi,us_aqi,pm2_5,pm10,ozone',
  })

  const res = await fetch(`https://air-quality-api.open-meteo.com/v1/air-quality?${params}`)
  if (!res.ok) throw new Error(`Air Quality API error: ${res.status}`)
  return res.json()
}
